scripts/package/fpsci_packager_generator.py

```diff
@@ -0,0 +1,125 @@
+
+    # main loop, need to ignore lines before the file list begins
+    beforeFiles = True
+    for line in log.readlines():
+        # In the file list, add this file to what we want to copy
+        if not beforeFiles and line != '\n':
+            filename = line.strip()
+            basename = filename.split('/')[-1]
+
+            # Exclude the lists of files we don't want to package
+            if basename in excludeList or basename in secondaryExcludeList:
+                continue
+            
+            # Absolute path given
+            if filename[1:].startswith(':/'):
+                for path in basePath:
+                    dest = filename.find(path)
+                    if dest > 0:
+                        dest += len(path) + 1
+                        additionalPath = filename[dest:-len(basename)]
+                        # if this file is inside a compressed structure, skip copying it
+                        if additionalPath.find('.pk3') > 0 or additionalPath.find('.zip') > 0:
+                            break
+
+                        # Make paths generic
+                        filename = filename.replace(g3dPath, "$g3d").replace(runPath, ".")
+
+                        # Write commands to file
+                        lineSet.add('mkdir -p ' + distPath + additionalPath + '\n')
+                        lineSet.add('cp ' + filename + ' ' + distPath + additionalPath + '\n')
+
+            # file is in the PATH, need to find it
+            else:
+                lineSet.add('cp `which ' + filename + '` ' + distPath + '\n')
+
+        # Find where the filenames start
+        if line.startswith('    ###    Files Used    ###'):
+            beforeFiles = False
+    log.close()
+
+    # Write the set of commands in sorted order
+    with open(outputScript, 'w') as packageScript:
+        # Script header and timestamp
+        packageScript.writelines(['#!/bin/bash\n', '# Autogenerated ' + now.strftime('%Y %m %d %H:%M:%S\n')])
+        for line in sorted(lineSet, reverse=True):
+            packageScript.write(line)
+
+    # Write readme
+    readmeString = "Welcome to First Person Science! We're glad you're joining us in studying first person aiming.\n\n"
+    readmeString += "This distribution represents the %s build of FPSci generated on %s.\n\n"%(currentRelease, now.strftime('%B %d, %Y'))
+    readmeString += "You can find documentation on how to create and customize your experiments here: https://github.com/NVlabs/abstract-fps/tree/%s/docs\n\n"%(currentRelease)
+    readmeString += "In order to use this build, you may need to install the x64 version (and possibly the x86 version) of the Microsoft Visual C++ Redistributable found at this link available here: https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads\n\n"
+    readmeString += "If you're interested, the source for this version is available here: https://github.com/NVlabs/abstract-fps/tree/%s\n"%(currentRelease)
```

Comment:
IMO a multi-line string would be more confusing for some of this (particularly when injecting variables). It would be cleaner for the other parts, but since variables show up on every line except 2 disconnected lines, I thought this format was more clear.